subprojects/logging/src/test/groovy/org/gradle/internal/logging/sink/ProgressLogEventGeneratorTest.groovy

```diff
@@ -17,468 +17,145 @@
 
 package org.gradle.internal.logging.sink
 
+import org.gradle.api.logging.LogLevel
+import org.gradle.internal.SystemProperties
 import org.gradle.internal.logging.OutputSpecification
+import org.gradle.internal.logging.events.EndOutputEvent
+import org.gradle.internal.logging.events.OperationIdentifier
 import org.gradle.internal.logging.events.OutputEventListener
-import org.gradle.internal.logging.events.StyledTextOutputEvent
+import org.gradle.internal.logging.events.ProgressCompleteEvent
+import org.gradle.internal.logging.events.ProgressStartEvent
+import org.gradle.internal.progress.BuildOperationType
+import org.gradle.util.MockTimeProvider
 
 class ProgressLogEventGeneratorTest extends OutputSpecification {
```

Comment:
I didn't see a test case that covered a more interleaved flow of events:

`:a` start event
`:a` renderable event
`:b` start event
`:a` renderable event
`:b` renderable event
`:b` complete event
`:a` complete event